src/coreclr/jit/valuenum.cpp

```diff
@@ -8494,6 +8494,65 @@ void Compiler::fgValueNumberSsaVarDef(GenTreeLclVarCommon* lcl)
     }
 }
 
+//----------------------------------------------------------------------------------
+// fgGetFieldSeqAndAddress: Try to obtain a constant address with a FieldSeq from the
+//    given tree. It can be either INT_CNS or e.g. ADD(INT_CNS, ADD(INT_CNS, INT_CNS))
+//    tree where only one of the constants is expected to have a field sequence.
+//
+// Arguments:
+//    tree     - tree node to inspect
+//    pAddress - [Out] resulting address with all offsets combined
+//    pFseq    - [Out] field sequence
+//
+// Return Value:
+//    true if the pattern was recognized and a new VN is assigned
+//
+static bool fgGetFieldSeqAndAddress(GenTree* tree, ssize_t* pAddress, FieldSeq** pFseq)
```

Comment:
Is the purpose of this function different from `GenTree::IsFieldAddr`?

Reply:
Sort of but different, e.g. IsFieldAddr will give up on `ADD(cns, static-field-handle)`

Reply:
:+1: Looks good with the new name.